lib/TPP/RewriteConvsToMatmulOrBrgemm.cpp

```diff
@@ -240,15 +240,14 @@ struct BlockConv2DNchwFchw : OpRewritePattern<linalg::Conv2DNchwFchwOp> {
   SmallVector<int64_t> blockingFactors;
 };
 
-// Interchange iterator in a linalg.generic marked as
-// 'tpp.BlockedConv2DNchwFchwOp' to expose a GEMM operation.
+// Interchange a blocked convolutions to expose a linalg.matmul.
 struct InterchangeIteratorsConv2DNchwFchw
     : OpRewritePattern<linalg::GenericOp> {
   using OpRewritePattern::OpRewritePattern;
 
   LogicalResult matchAndRewrite(linalg::GenericOp genericOp,
                                 PatternRewriter &rewriter) const override {
-    if (!tpp::utils::isMarkedWithTpp(genericOp, "tpp.BlockedConv2DNchwFchwOp"))
+    if (!linalgx::utils::isBlockedConvolution(genericOp))
```

Comment:
Why is this in `linalgx` namespace? We want to remove that, not add more functionality there.

You could just add that to the `linalg` namespace, and "extend" it that way. No need for an `x` at the end.

But only if we really want to upstream that, or if there's no TPP logic in it. Otherwise, it should be local and in some tpp namespace (as it was before).

Reply:
I feel linalgx is a better namespace than tpp because these methods works at linalg. I prefer not to pollute linalg namespace.

Reply:
We map both blocked and non-blocked convs. So the first run also map to linalg even if it does not block the conv.

Reply:
I don't like the namespace but it doesn't matter much. If we want this to be linalg-only, then perhaps we should upstream those helpers?

Reply:
Let's keep them in-house a little longer. Don't think are relevant for upstream at the moment.